libs/algo/odc/algo/io.py

```diff
@@ -281,35 +297,51 @@ def load_enum_filtered(
     """
 
     def native_op(xx: xr.Dataset) -> xr.Dataset:
-        _xx = enum_to_bool(xx[band], categories)
-        return xr.Dataset(
+        if len(np.array(filters).shape) < 3:
```

Comment:
For all the change below, I don't think we should hack inside, but consider to change the interface of this function. Currently it accepts a union of values and convert the input band to a boolean `xr.DataArray`. It could accepts a list of values and convert each of them to a boolean `xr.Dataset` with mask/s as variable/s. Again we should avoid the list for `dask`'s sake, since it'll introduce unnecessary troubles in the future development.